File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        if len(nums) == 0:
            return 0
        minimum = min(nums)
        maximum = max(nums)
        cache = {}
        for i in range(minimum,maximum + 1):
            cache[i] = 0
        result, curr = 0, 0

        for num in nums:
            cache[num] += 1


        vals = list(cache.values())
        for num in vals:
            if num > 0:
                curr += 1
            else:
                result = max(result, curr)
                curr = 0

        return max(result, curr)
        

File: Data Structures & Algorithms/maximum-product-subarray/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxProduct(nums: number[]): number {
        let curr: number = 1;

        let res: number = -Infinity
        for (let i = 0; i < nums.length; i++) {
            if (nums[i] === 0) {
                res = Math.max(0,res)
                curr = 1;
                continue;
            }
            curr *= nums[i]
            res = Math.max(curr,res)
        }
        
        curr = 1
        for (let i = nums.length - 1; i >= 0; i--) {
            if (nums[i] === 0) {
                res = Math.max(0,res)
                curr = 1;
                continue;
            }
            curr *= nums[i]
            res = Math.max(curr,res)
        }
        return res
    }
}
